pkg/cli/demo.go

```diff
@@ -189,6 +235,29 @@ func setupTransientServers(
 	}
 	urlStr := url.String()
 
+	// Start up the update check loop.
+	// We don't do this in (*server.Server).Start() because we don't want it
+	// in tests.
+	if shouldEnableTelemetry() {
+		s.PeriodicallyCheckForUpdates(ctx)
+		// If we allow telemetry, then also try and get an enterprise license for the demo.
+		if cliCtx.isInteractive {
+			db, err := gosql.Open("postgres", urlStr)
+			if err != nil {
+				return ``, ``, cleanup, err
+			}
+			// Perform license acquisition asynchronously to avoid delay in cli startup.
+			go func() {
+				defer db.Close()
+				err := getAndApplyLicense(db)
+				if err != nil {
```

Comment:
1. if the error is non-`nil`, report as a real error (this means the SQL connection has failed or one of the SQL statements was not intepreted correctly. This is a hard error)
2. if the error is `nil` but the boolean is false, the license acquisition has failed and then you can print the warning.